app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>FarMarRails -- adh+js/master</title>
+  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+  <link href='http://fonts.googleapis.com/css?family=Open+Sans:400,300,700,600' rel='stylesheet' type='text/css'>
+  <%= csrf_meta_tags %>
+</head>
+<body>
+  <nav>
+    <%= link_to "All Markets", "/markets", title: "All Markets" %>
+
+    <% id = params[:id] || params[:market_id] || params[:vendor_id] || false %>
+    <%= link_to id ? "Log Out" : "Log In", "/", title: id ? "Log out of your account." : "Log in to your account." %>
```

Comment:
Try sticking this logic in your controller or in a helper method. Both lines 14 and 15 is pretty difficult to parse. You might try rewriting it on multiple lines to help make it easier to read?